src/core/middleware.py

```diff
@@ -0,0 +1,35 @@
+from django.conf import settings
+from django.views.decorators.cache import patch_cache_control
+from django.utils.cache import patch_vary_headers
+
+class DownstreamCaching(object):
```

Comment:
for context: as the name suggests, this only adds headers so that CDNs can cache the responses; but backends should never locally cache the responses themselves